### PR TITLE
Add ConfigureAwait(false) when retrieving metadata. #320

### PR DESCRIPTION
**Pull request by brentschmaltz**
Add ConfigureAwait(false) when retrieving metadata. #320

### CODE REVIEW

File: src/Microsoft.IdentityModel.Protocols/Configuration/ConfigurationManager.cs [hunk 1]
```diff
@@ -152,7 +152,7 @@ public TimeSpan RefreshInterval
         /// <remarks>If the time since the last call is less than <see cref="AutomaticRefreshInterval"/> then <see cref="IConfigurationRetriever{T}.GetConfigurationAsync"/> is not called and the current Configuration is returned.</remarks>
         public async Task<T> GetConfigurationAsync()
         {
-            return await GetConfigurationAsync(CancellationToken.None);
+            return await GetConfigurationAsync(CancellationToken.None).ConfigureAwait(false);
```
Comment (PinpointTownes) [inline src/Microsoft.IdentityModel.Protocols/Configuration/ConfigurationManager.cs]: nit: you can remove `async` and directly return the task here, without having to use `ConfigureAwait(false);`.

**Comment by tushargupta51** [comment]
@brentschmaltz do we need configureAwait in FileDocumentRetriever also?

**Comment by tushargupta51** [comment]
:shipit:

**Comment by brentschmaltz** [comment]
Fixed issue #321 (Configure Await)

**Pull request closed without merging**